bin/check-graphite-hosts.rb

```diff
@@ -0,0 +1,104 @@
+#! /usr/bin/env ruby
+#
+#   check-graphite-hosts
+#
+# DESCRIPTION:
+#   This plugin checks the number of hosts within graphite that are sending
+#   data, and alerts if it is below a given threshold
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: openssl
+#
+# USAGE:
+#   #YELLOW
+#
+# NOTES:
+#
+# LICENSE:
+#   Copyright 2014 Sonian, Inc. and contributors. <support@sensuapp.org>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'json'
+require 'open-uri'
+require 'openssl'
+
+require 'sensu-plugins-graphite/graphite_proxy/options'
+require 'sensu-plugins-graphite/graphite_proxy/proxy'
+
+class CheckGraphiteHosts < Sensu::Plugin::Check::CLI
+
+  include SensuPluginsGraphite::GraphiteProxy::Options
+
+  # Run checks
+  def run
+    if config[:help]
+      puts opt_parser if config[:help]
+      exit
+    end
+
+    proxy = SensuPluginsGraphite::GraphiteProxy::Proxy.new(config)
+    begin
+      results = proxy.retrieve_data!
+      check_age(results) || check(:critical, results) || check(:warning, results)
+  
+      ok("#{name} value (#{hosts_with_data(results)}) okay")
+    rescue SensuPluginsGraphite::GraphiteProxy::ProxyException => e
+      puts e.backtrace
+      unknown e.message
+    end
+  end
+
+  # name used in responses
+  def name
+    base = config[:name]
+    @formatted ? "#{base} (#{@formatted})" : base
+  end
+
+  # Check the age of the data being processed
+  def check_age(results)
```

Comment:
This type of validation doesn't make sense for this check – it is there to make sure that data you are checking for a metric is recent enough, but the very nature of this check is that we are looking at "how many hosts are there in the last 15 minutes" so if one of those has no data for the last 10 minutes we don't care.